pkg/client/cli/cmd_gatherlogs.go

```diff
@@ -0,0 +1,255 @@
+package cli
+
+import (
+	"archive/zip"
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/telepresenceio/telepresence/rpc/v2/connector"
+	"github.com/telepresenceio/telepresence/rpc/v2/manager"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/cliutil"
+	"github.com/telepresenceio/telepresence/v2/pkg/filelocation"
+)
+
+type gatherLogsArgs struct {
+	outputFile     string
+	daemons        string
+	trafficAgents  string
+	trafficManager bool
+}
+
+func gatherLogsCommand() *cobra.Command {
+	gl := &gatherLogsArgs{}
+	cmd := &cobra.Command{
+		Use:   "gather-logs",
+		Args:  cobra.NoArgs,
+		Short: "Gather logs from traffic-manager, traffic-agent, user and root daemons, and export them into a zip file.",
+		Long: `Gather logs from traffic-manager, traffic-agent, user and root daemons,
+and export them into a zip file. Useful if you are opening a Github issue or asking
+someone to help you debug Telepresence.`,
+		Example: `Here are a few examples of how you can use this command:
+# Get all logs and export to a given file
+telepresence gather-logs -o /tmp/telepresence_logs.zip
+
+# Get all logs for the daemons only
+telepresence gather-logs --traffic-agents=False --traffic-manager=False
+
+# Get all logs for pods that have "echo-easy" in the name, useful if you have multiple replicas
+telepresence gather-logs --traffic-manager=False --traffic-agents=echo-easy
+
+# Get all logs for a specific pod
+telepresence gather-logs --traffic-manager=False --traffic-agents=echo-easy-6848967857-tw4jw     
+
+# Get logs from everything except the daemons
+telepresence gather-logs --daemons=False
+`,
+
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return gl.gatherLogs(cmd, cmd.Context(), cmd.OutOrStdout())
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&gl.outputFile, "output-file", "o", "", "The file you want to output the logs to.")
+	flags.StringVar(&gl.daemons, "daemons", "all", "The daemons you want logs from: all, root, user, False")
+	flags.BoolVar(&gl.trafficManager, "traffic-manager", true, "If you want to collect logs from the traffic-manager")
+	flags.StringVar(&gl.trafficAgents, "traffic-agents", "all", "Traffic-agents to collect logs from: all, name substring, False")
+	return cmd
+}
+
+// gatherLogs gets the logs from the daemons (daemon + connector) and creates a zip
+// file with their contents.
+func (gl *gatherLogsArgs) gatherLogs(cmd *cobra.Command, ctx context.Context, stdout io.Writer) error {
```

Comment:
could we add some scout reporting to this? my thinking is if a user is running gather-logs it's probably an indicator of some kind of issue, and it'd be good to have that represented in our metrics

Reply:
also good to get metrics / usage of the command in general